packages/x-data-grid-premium/src/components/GridAggregationHeader.tsx

```diff
@@ -42,11 +42,10 @@ const GridAggregationFunctionLabel = styled('div', {
   return {
     fontSize: theme.typography.caption.fontSize,
     lineHeight: theme.typography.caption.fontSize,
+    color: theme.palette.text.secondary,
     position: 'absolute',
-    bottom: 4,
-    fontWeight: theme.typography.fontWeightMedium,
-    color: (theme.vars || theme).palette.primary.dark,
-    textTransform: 'uppercase',
+    top: '50%',
```

Comment:
Perhaps we can move the label down a bit to avoid a potential collision with the column header text?
WDYT @KenanYusuf ?

<img width="555" alt="Screenshot 2024-07-26 at 15 36 12" src="https://github.com/user-attachments/assets/83f2592f-a35c-43fe-96d5-f380b1d9f675">

> In case of a custom aggregation label

Reply:
I agree, it could be moved down more. 

The problem with that is the available space on `density="compact"`

<img width="849" alt="Screenshot 2024-07-26 at 14 42 00" src="https://github.com/user-attachments/assets/4b3929ff-56eb-4b83-98f7-b890067ea498">

As it stands, the aggregation label is already touching the bottom border when the label has a descender e.g. `avg`. 

I would propose that we create a separate set of density factor values for column headers to ensure that our header cells always have enough space for the various elements including the aggregation label. What do you think @cherniavskii 

We have an issue open already to look into this problem https://github.com/mui/mui-x/issues/5274, but I could do it as part of this PR.

Reply:
Made a separate PR for the above #14022